control/available_plugin.go

```diff
@@ -457,6 +458,10 @@ func (ap *availablePlugins) publishMetrics(contentType string, content []byte, p
 		return []error{serror.New(ErrPoolNotFound, map[string]interface{}{"pool-key": key})}
 	}
 
+	if !pool.HasSubscription(taskID) {
+		return []error{serror.New(ErrBadTaskId, map[string]interface{}{"task-id": taskID})}
+	}
+
```

Comment:
How was the case where a plugin is started and added to a pool but not subscribed to handled previously? I think the subscription is what kicks off adding a plugin to a pool. I'd be curious what sequence of events would have you end up with a pool for a plugin that doesn't have a valid subscription to it?